### ISSUE TITLE
Version 1.2 crashes almost all the times

### ISSUE THREAD

**Issue by santi-ifontana** [thayakawa-gh/SlackLogViewer#21]
Hello.

I am working on a project trying to visualize the export from Slack. The .Zip file is around 300 MB zipped.
The version 1.1 works fine most of the time but only has Channels while 1.2 has DM and other messages.

The last version 1.2 Alpha-2 won't even open complaining about missing users in users.json while the other ones crash while opening the .Zip or if it opens, while navigating.

How can I help you fix those bugs? Please let me know.

Thanks.

**Comment by thayakawa-gh** [comment]
Hi, @santi-infontana. Thank you for your report.
To determine if the issue is caused by corruption in users.json, please confirm that "missing users" IDs are listed in users.json. Additionally, please let me know how you exported the zip file (e.g., slackdump, Slack's official functionality, etc.). Based on your report, it is difficult to say whether the crash is due to bugs in SlackLogViewer.

**Comment by santi-ifontana** [comment]
@thayakawa-gh Hey. Thank you for your reply. Yes, sorry for my vague description.

So, the Slack export is directly from Slack support (Slack's official functionality).

Trying to open the same folder or zip file, with the different versions:

SlackLogViewer ver 1.2.Alpha-2
<img width="1440" alt="image" src="https://github.com/thayakawa-gh/SlackLogViewer/assets/165914151/ac06ba82-aea0-4672-8444-e734c34469ca">

SlackLogViewer ver 1.2.Alpha-0 and Alpha-1 do not show that error, although I never managed to get Alpha-1 to open and not crash before. Alpha-0 after trying a few times is able to open, but very unstable and ends up crashing.

**Comment by santi-ifontana** [comment]
The uncompressed .zip is pretty huge regarding size and number of files as shown: 
<img width="358" alt="Screenshot 2024-06-26 at 19 25 34" src="https://github.com/thayakawa-gh/SlackLogViewer/assets/165914151/2c148f98-abf0-4653-880e-764034f640b5">

**Comment by euphorah** [comment]
i'm experiencing similar issue with export from [slackdump v2.5.10](https://github.com/rusq/slackdump/releases/tag/v2.5.10)

upon opening slackdump zip & folders with following outcomes
[v1.2.Alpha-2](https://github.com/thayakawa-gh/SlackLogViewer/releases/tag/v1.2.Alpha-2) = crashes with the user id not found
[v1.2.Alpha-1](https://github.com/thayakawa-gh/SlackLogViewer/releases/tag/v1.2.Alpha-1) & [v1.2.Alpha-0](https://github.com/thayakawa-gh/SlackLogViewer/releases/tag/v1.2.Alpha-0) = crashes with no error
[v1.1.0](https://github.com/thayakawa-gh/SlackLogViewer/releases/tag/v1.1.0) = works fine

export is greater than 2GB but mostly media files

**Comment by santi-ifontana** [comment]
> Sorry for my late reply. Please try [this test build for windows](https://github.com/thayakawa-gh/SlackLogViewer/actions/runs/10548290850/artifacts/1851962363) ([or macOS](https://github.com/thayakawa-gh/SlackLogViewer/actions/runs/10548290849/artifacts/1851963557)).

Hey! This version does the trick, no crashing! Although the app really seems to struggle with the big backup file, but that is ok.

I saw the changes you made to the code, basically, there are some users in the .json that are missing and on that error, you were throwing a fatal error and now instead just giving a fake user back, isn't it?

**Comment by thayakawa-gh** [comment]
You are correct. The changes simply make SlackLogViewer tolerate missing user information. Slack export files often have missing data that should be present. Your report revealed that even user information could be missing, so I fixed the code to handle it.

**Comment by santi-ifontana** [comment]
Thanks for the fix and the explanation. For me, this issue could be closed. I don't know about @euphorah if that fixed it for him too.

Regards

**Comment by euphorah** [comment]
> Thanks for the fix and the explanation. For me, this issue could be closed. I don't know about @euphorah if that fixed it for him too.
> 
> Regards

my apologies i've got a lot of cognitive issues so might not be able to keep up with things

the test build resolves this issue however am unable to load some channels due to 

"Error : Loading of the export files was interrupted due to some formatting errors. See errorlog.txt for details.

key "bot_id" not found in the 18th message of "channel3/2024-06-19.json".
key "bot_id" not found in the 0th message of "channel3/2024-06-20.json".

i've attached deidentified sections of those files that error with X for letters and 0 for numbers hopefully it's still useful

[18th-message_channel3_2024-06-19.json.txt](https://github.com/user-attachments/files/16825540/18th-message_channel3_2024-06-19.json.txt)

[0th-message_channel3_2024-06-20.json.txt](https://github.com/user-attachments/files/16825541/0th-message_channel3_2024-06-20.json.txt)

both messages display fine in [v1.1.0](https://github.com/thayakawa-gh/SlackLogViewer/releases/tag/v1.1.0)

![Screenshot 2024-09-01 032457](https://github.com/user-attachments/assets/45ce5307-c9b8-4b4e-af5d-34988c8e02b7)

**Comment by thayakawa-gh** [comment]
Thank you for your attachments, they were very helpful for me.
The format of the Slack export files is as inconsistent as ever. Could you try [this one for windows](https://github.com/thayakawa-gh/SlackLogViewer/actions/runs/10655242372/artifacts/1879550006) or [macOS](https://github.com/thayakawa-gh/SlackLogViewer/actions/runs/10655242370/artifacts/1879548688)?
